lib/KettleRequest.js

```diff
@@ -295,6 +328,8 @@ kettle.request.initiateHandleRequest = function (request, rootSequence) {
     fluid.log("Kettle server allocated request object with type ", request.typeName);
     var requestSequence = kettle.middleware.getHandlerSequence(request, "requestMiddleware");
     var fullSequence = rootSequence.concat(requestSequence).concat([kettle.request.handleRequestTask]);
+    // FIXME: we *think* the below line could just be replaced with
+    // fluid.promise.sequence(fullSequence, request);
     var handleRequestPromise = kettle.request.sequenceRequest(fullSequence, request);
```

Comment:
The suggested refactoring does not work, nor does `var handleRequestPromise = fluid.promise.sequence(fullSequence, reqest);`.

Reply:
I wonder why. The comment does seem to be explicit that this is the desired behaviour - "The yielded promise will only resolve if the request's `handlerPromise` has resolved." - but the reasoning is no longer clear.

Reply:
Ah, I know why this is - the yielded value from fluid.promise.sequence is off - rather than returning the "right" success payload, it instead returns an array of all the resolves. The docs are not clear on this point https://docs.fluidproject.org/infusion/development/PromisesAPI.html#fluidpromisesequencesources-options - we should fix this as well as the unclarity that @simonbates noted some time ago about the exact semantics of "sequencing"

Reply:
Fixing the Infusion Promises API seems like a highly breaking change. How do we fetch the desired value out of the result of fluid.promise.sequence?

Reply:
I'm not suggesting we fix it, just the docs - as well as the docs here to explain the reasoning more clearly

Reply:
Could you draft what you think the docs here should be? It sounds like you have rediscovered why `kettle.request.sequenceRequest` is expressed the way it is, but I'm not there yet :)

Reply:
I spent a little time a while ago making an update to the documentation for `fluid.promise.sequence` but somehow got distracted and didn't make the pull request. Sorry about that! There's maybe something here that could be salvaged?

https://github.com/fluid-project/infusion-docs/compare/master...simonbates:FLUID-6374

Reply:
Thanks, @simonbates - highly useful! Will try and work it in